python/llm/src/bigdl/llm/transformers/models/falcon.py

```diff
@@ -0,0 +1,214 @@
+
+    key_cache = key_cache_storage.as_strided((batch_size*num_heads,
+                                            current_length, head_dim),
+                                            key_cache_storage.stride(),
+                                            storage_offset=0)
+    value_cache = value_cache_storage.as_strided((batch_size*num_heads,
+                                                current_length, head_dim),
+                                                value_cache_storage.stride(),
+                                                storage_offset=0)
+    return key_cache, value_cache
+
+
+def append_kv_cache(cache_k, cache_v, key_states, value_states):
+    new_size = (cache_k.size(0),
+                cache_k.size(1) + key_states.size(1),
+                cache_k.size(2))
+    new_cache_k = cache_k.as_strided(new_size, cache_k.stride(), storage_offset=0)
+    new_cache_k[:, cache_k.size(1):cache_k.size(1) + key_states.size(1), :] = key_states
+    new_cache_v = cache_v.as_strided(new_size, cache_v.stride(), storage_offset=0)
+    new_cache_v[:, cache_v.size(1):cache_k.size(1) + key_states.size(1), :] = value_states
+    return new_cache_k, new_cache_v
+
+
+KV_CACHE_ALLOC_BLOCK_LENGTH = 256
+
+def falcon_attention_forward(
+        self,
+        hidden_states: torch.Tensor,
+        alibi: Optional[torch.Tensor],
+        attention_mask: torch.Tensor,
+        layer_past: Optional[Tuple[torch.Tensor, torch.Tensor]] = None,
+        head_mask: Optional[torch.Tensor] = None,
+        use_cache: bool = False,
+        output_attentions: bool = False,
+    ):
+        fused_qkv = self.query_key_value(hidden_states)  # [batch_size, seq_length, 3 x hidden_size]
+        num_kv_heads = self.num_heads if self.new_decoder_architecture else self.num_kv_heads
+        # 3 x [batch_size, seq_length, num_heads, head_dim]
+        (query_layer, key_layer, value_layer) = self._split_heads(fused_qkv)
+
+        batch_size, query_length, _, _ = query_layer.shape
+
+        query_layer = query_layer.transpose(1, 2).reshape(batch_size * self.num_heads, query_length, self.head_dim)
+        key_layer = key_layer.transpose(1, 2).reshape(
+            batch_size * num_kv_heads,
+            query_length,
+            self.head_dim,
+        )
+        value_layer = value_layer.transpose(1, 2).reshape(batch_size * num_kv_heads, query_length, self.head_dim)
+
+        past_kv_length = 0 if layer_past is None else layer_past[0].shape[1]
+        query_layer, key_layer = self.maybe_rotary(query_layer, key_layer, past_kv_length)
+
+        _, kv_length, _ = key_layer.shape
+
+        bsz, q_len, _ = hidden_states.size()
+        device = hidden_states.device
+        if layer_past is not None:
+            # reuse k, v, self_attention
+            cache_k = layer_past[0]
+            cache_v = layer_past[1]
+            if cache_k.stride()[0] / bsz <= cache_k.size(1) * cache_k.size(2):
+                # allocate new
```

Comment:
```python
if device.type == 'xpu':
     torch.xpu.empty_cache()
```
See https://github.com/intel-analytics/BigDL/blob/main/python/llm/src/bigdl/llm/transformers/models/llama.py#L132-L133

Shall we add this logic to the `create_kv_cache` method? @rnwang04 @yangw1234

Reply:
Yeah， we can move this logic to the beginning of `create_kv_cache`